zqd/space/config.go

```diff
@@ -55,26 +74,133 @@ func (c config) subspaceIndex(id api.SpaceID) int {
 }
 
 // loadConfig loads the contents of config.json in a space's path.
-func loadConfig(spacePath string) (config, error) {
+func loadConfig(spaceURI iosrc.URI) (config, error) {
 	var c config
-	path := filepath.Join(spacePath, configFile)
-	if err := fs.UnmarshalJSONFile(path, &c); err != nil {
+	p := spaceURI.AppendPath(configFile)
+	r, err := iosrc.NewReader(p)
+	if err != nil {
+		return c, err
+	}
+	data, err := ioutil.ReadAll(r)
+	if err != nil {
+		r.Close()
+		return c, err
+	}
+	if err := r.Close(); err != nil {
+		return c, err
+	}
```

Comment:
It's fine to ignore a Close error from a reader.
```suggestion
	defer r.Close()
	data, err := ioutil.ReadAll(r)
	if err != nil {
		return c, err
	}
```